datafusion/expr/src/expr.rs

```diff
@@ -1175,26 +1175,26 @@ impl Exists {
 
 /// User Defined Aggregate Function
 ///
-/// See [`udaf::AggregateUDF`] for more information.
+/// See [`crate::udaf::AggregateUDF`] for more information.
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
 pub struct AggregateUDF {
     /// The function
-    pub fun: Arc<udaf::AggregateUDF>,
+    pub fun: Arc<crate::AggregateUDF>,
     /// List of expressions to feed to the functions as arguments
     pub args: Vec<Expr>,
     /// Optional filter
     pub filter: Option<Box<Expr>>,
     /// Optional ORDER BY applied prior to aggregating
-    pub order_by: Option<Vec<Expr>>,
+    pub order_by: Vec<Sort>,
```

Comment:
I understand why the Optional was removed (that's the purpose of this PR). I don't yet understand why Expr was replaced with Sort thought.

Reply:
This was a mistake during the batch replacement process, but since this struct has never been used in the project, it did not cause any exceptions. I'll change it back.

Reply:
And we can open another issue to discuss whether it is more reasonable to change it to the Sort type, because other order_by fields are defined as the Sort type. Or we can directly delete this unused struct.

Reply:
if it's unused, let's delete it, prefferrably in a separate PR

Reply:
OK. Should it be before or after this PR?